### ISSUE TITLE
Props missing in props validation when defined as flow types in a separate file

### ISSUE THREAD

**Issue by ca057** [yannickcr/eslint-plugin-react#1622]
From time to time I'm in the need of sharing my Flow type definitions for my props between different components. To do so, I extracted them to a separate `types.js`-file to simplify the imports on the other side.

Nevertheless, I get problems with the original component and the prop validation.

E.g., when having a file `types.js`:
```javascript
// @flow
export type Props = {
  foo: string
};
```
and a component in my `index.js`:
```javascript
// @flow
import * as React from 'react';
import type { Props } from './types';

class C extends React.Component<Props> {
  props: Props;
  
  render() {
    return <p>{this.props.foo}</p>;
  }
}
```
eslint reports an error that `foo` is missing in props validation, due to the `react/prop-types`-rule.

Is this possible with the current plugin/eslint setup? Am I missing some configuration? Is this a feature which might be imported in a future versions or do you have no plans to do so?

Thanks for your help!

My configuration:
```json
{
  "devDependencies":
    "eslint-plugin-react": "7.3.0",
    "flow-bin": "^0.61.0",
  }
}
```

**Comment by ljharb** [comment]
It’s not practically possible to track type deps across files. Typically, Props needs to always be defined in the same file, although shapes can be shared.

**Comment by ca057** [comment]
Thanks for your reply!

**Comment by stomvi** [comment]
Is this the reason that if the `props` is checked using flow annotation and the `Props` type is imported from a separate file, `eslint-plugin-react` will show `react/default-props-match-prop-types` error?

**Comment by ljharb** [comment]
I believe so, yes.
